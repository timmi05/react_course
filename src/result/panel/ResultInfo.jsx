import React from 'react';

export const ResultInfo = (props) => {

    const { count, genre } = props;
    const massage = count ? `${ count } movie found` : genre ? `Film by ${ genre } genre` : '';

    return (
        <span className='result-info'>
        { massage }
        </span>
    );
}

export default ResultInfo