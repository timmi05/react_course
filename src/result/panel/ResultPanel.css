.result-panel{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 25px 30px 25px;
    font-size: 20px;
    margin: 0 40px;
}
