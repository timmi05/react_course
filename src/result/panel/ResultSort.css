.result-sort{
    display: flex;
}

.result-sort-button{
    font-size: 15px;
    padding: 10px 20px;
}

.result-sort-label{
    padding-right: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}