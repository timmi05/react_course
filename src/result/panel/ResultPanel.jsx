import React from 'react';
import { ResultInfo } from './ResultInfo';
import { ResultSort } from './ResultSort';
import './ResultPanel.css'

export const ResultPanel = props => {
    const { productSelected, genre, data } = props;
    return(
     <div className='result-panel'>
         <ResultInfo
         productSelected = {productSelected}
         genre = { genre }
         data = { data }
         />
         < ResultSort
         productSelected = { productSelected }
         />
    </div>
    )
};
