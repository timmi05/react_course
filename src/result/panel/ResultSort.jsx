import React, { useState } from 'react';
import { Button } from '../../component/Button'
import './ResultSort.css'

export const ResultSort = props => {

    const { productSelected } = props;
    const [disabled, setDisabled] = useState(false);

    const SortByReleaseDate = () => {
        setDisabled(true);
    };

    const SortByRating = () => {
        setDisabled(false);
    };

    if ( productSelected )
    return(
        <a/>
    );

    return (
        <span className='result-sort'>
        <label className='result-sort-label'>SORT BY </label>
            <div className='result-sort-buttons'>
                <Button
                label = 'RELEASE DATE'
                classes = 'result-sort-button result-sort-left-button'
                onClick = { SortByReleaseDate }
                disabled = { disabled }
                />
                <Button
                label = 'RATING'
                classes = 'result-sort-button result-sort-right-button'
                onClick = { SortByRating }
                disabled = { !disabled }
                />
            </div>
        </span>
      );
};
