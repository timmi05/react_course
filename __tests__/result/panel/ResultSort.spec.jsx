import React from 'react';
import { shallow, mount } from 'enzyme';
import { ResultSort } from '../../../src/result/panel/ResultSort';

describe('ResultPanel rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<ResultSort/>);
        expect(component).toMatchSnapshot();
    });

    it('renders if productSelected is true', () => {
        const component = shallow(<ResultSort
        productSelected = { true }
        />);
        expect(component).toMatchSnapshot();
    });

    it('renders if productSelected is true', () => {
        const component = shallow(<ResultSort
        productSelected = { false }
        />);
        expect(component).toMatchSnapshot();
    });

    it('simulate click on button by RELEASE DATE', () => {
        const component = mount(<ResultSort/>);
        component.find({label: 'RELEASE DATE'}).simulate('click');
        expect(component.find({className:'button result-sort-button result-sort-left-button'})
            .prop('disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });

    it('simulate click on button by RATING', () => {
        const component = mount(<ResultSort/>);
        component.find({label: 'RELEASE DATE'}).simulate('click');
        component.find({label: 'RATING'}).simulate('click');
        expect(component.find({className:'button result-sort-button result-sort-right-button'})
            .prop('disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });
});
