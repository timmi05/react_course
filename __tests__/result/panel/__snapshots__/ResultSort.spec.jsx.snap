// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPanel rendering renders correctly 1`] = `
<span
  className="result-sort"
>
  <label
    className="result-sort-label"
  >
    SORT BY 
  </label>
  <div
    className="result-sort-buttons"
  >
    <Button
      classes="result-sort-button result-sort-left-button"
      disabled={false}
      label="RELEASE DATE"
      onClick={[Function]}
    />
    <Button
      classes="result-sort-button result-sort-right-button"
      disabled={true}
      label="RATING"
      onClick={[Function]}
    />
  </div>
</span>
`;

exports[`ResultPanel rendering renders if productSelected is true 1`] = `<a />`;

exports[`ResultPanel rendering renders if productSelected is true 2`] = `
<span
  className="result-sort"
>
  <label
    className="result-sort-label"
  >
    SORT BY 
  </label>
  <div
    className="result-sort-buttons"
  >
    <Button
      classes="result-sort-button result-sort-left-button"
      disabled={false}
      label="RELEASE DATE"
      onClick={[Function]}
    />
    <Button
      classes="result-sort-button result-sort-right-button"
      disabled={true}
      label="RATING"
      onClick={[Function]}
    />
  </div>
</span>
`;

exports[`ResultPanel rendering simulate click on button by RATING 1`] = `
<ResultSort>
  <span
    className="result-sort"
  >
    <label
      className="result-sort-label"
    >
      SORT BY 
    </label>
    <div
      className="result-sort-buttons"
    >
      <Button
        classes="result-sort-button result-sort-left-button"
        disabled={false}
        label="RELEASE DATE"
        onClick={[Function]}
      >
        <button
          className="button result-sort-button result-sort-left-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          RELEASE DATE
        </button>
      </Button>
      <Button
        classes="result-sort-button result-sort-right-button"
        disabled={true}
        label="RATING"
        onClick={[Function]}
      >
        <button
          className="button result-sort-button result-sort-right-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          RATING
        </button>
      </Button>
    </div>
  </span>
</ResultSort>
`;

exports[`ResultPanel rendering simulate click on button by RELEASE DATE 1`] = `
<ResultSort>
  <span
    className="result-sort"
  >
    <label
      className="result-sort-label"
    >
      SORT BY 
    </label>
    <div
      className="result-sort-buttons"
    >
      <Button
        classes="result-sort-button result-sort-left-button"
        disabled={true}
        label="RELEASE DATE"
        onClick={[Function]}
      >
        <button
          className="button result-sort-button result-sort-left-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          RELEASE DATE
        </button>
      </Button>
      <Button
        classes="result-sort-button result-sort-right-button"
        disabled={false}
        label="RATING"
        onClick={[Function]}
      >
        <button
          className="button result-sort-button result-sort-right-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          RATING
        </button>
      </Button>
    </div>
  </span>
</ResultSort>
`;
